### PR TITLE
formly: improve id generator

### PR DESCRIPTION
**Pull request by Garfield-fr**
* Adds numerical position on key generation for OneOf.

Co-Authored-by: Bertrand Zuchuat <bertrand.zuchuat@rero.ch>
Co-Authored-by: Johnny Mariéthoz <johnny.mariethoz@rero.ch>

## Why are you opening this PR?

- Fix generate id on form field

## How to test?

- Check id on form

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Extracted translations?

### CODE REVIEW

File: projects/rero/ng-core/src/lib/record/editor/extensions.ts [hunk 1]
```diff
@@ -74,11 +74,15 @@ export function getKey(field: any): string {
...
   if (parentKey != null) {
-    let newId = parentKey;
+    const keyParams = [];
     if (field.key != null) {
-      newId = [parentKey, field.key].join('-');
+      keyParams.push(parentKey);
+      keyParams.push(field.key);
     }
-    return newId;
+    if (field.id) {
```
Comment (jma) [inline projects/rero/ng-core/src/lib/record/editor/extensions.ts:82]: I do not understand as the idea is to generate the id.

File: projects/rero/ng-core/src/lib/record/editor/extensions.ts [hunk 2]
```diff
@@ -74,11 +74,15 @@ export function getKey(field: any): string {
   }
 
   if (parentKey != null) {
-    let newId = parentKey;
+    const keyParams = [];
     if (field.key != null) {
-      newId = [parentKey, field.key].join('-');
+      keyParams.push(parentKey);
```
Comment (jma) [inline projects/rero/ng-core/src/lib/record/editor/extensions.ts:79]: why only if the field.key is defined? This should be done if the parentKey exists even if the field.key is undefined (oneOf)

File: projects/rero/ng-core/src/lib/record/editor/extensions.ts (hunk 1 above)
Comment (Garfield-fr) [reply projects/rero/ng-core/src/lib/record/editor/extensions.ts]: done

File: projects/rero/ng-core/src/lib/record/editor/extensions.ts (hunk 2 above)
Comment (Garfield-fr) [reply projects/rero/ng-core/src/lib/record/editor/extensions.ts]: done

**Pull request merged**